app/src/main/java/co/touchlab/kampkit/android/ui/Composables.kt

```diff
@@ -0,0 +1,155 @@
+package co.touchlab.kampkit.android.ui
+
+import androidx.compose.animation.Crossfade
+import androidx.compose.animation.core.FastOutSlowInEasing
+import androidx.compose.animation.core.TweenSpec
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Divider
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import co.touchlab.kampkit.android.R
+import co.touchlab.kampkit.db.Breed
+import co.touchlab.kampkit.models.DataState
+import co.touchlab.kampkit.models.ItemDataSummary
+
+@Composable
+fun DogList(breeds: List<Breed>, onItemClick: (Breed) -> Unit) {
+    LazyColumn {
+        items(breeds) { breed ->
+            DogRow(breed) {
+                onItemClick(it)
+            }
+            Divider()
+        }
+    }
+}
+
+@Composable
+fun DogRow(breed: Breed, onClick: (Breed) -> Unit) {
+    Row(
+        Modifier
+            .clickable { onClick(breed) }
+            .padding(10.dp)
```

Comment:
I took this from your commit @mrf7 , so thank you for that. But `clickable` and `padding` were in the opposite order as here. I noticed when I clicked on the item, the padding didn't get the ripple effect, but with `clickable` before `padding`, the ripple effect worked. And I really didn't expect that. I'd expect padding first would be the way to go, and then applying clickable would make everything, including the `padding`, `clickable`. Anyway, order matters in Compose Modifiers, and I learned that sometimes that order might not be what you'd expect.

Reply:
Yea i had the same realization. Might be a blog post's amount of content in the ordering of modifiers in compose since this one in particular seemed backwards